speechbrain/decoders/seq2seq.py

```diff
@@ -1551,156 +1856,7 @@ def forward_step(self, inp_tokens, memory, enc_states, enc_lens):
-        self.set_task_token(decoder_input_tokens[2])
-        self.set_timestamp_token(decoder_input_tokens[3])
-
-        # bos will be timestamp in our case.
-        self.decoder_input_tokens = [
-            self.bos_token,
-            self.language_token,
-            self.task_token,
-        ]
-
-    def reset_mem(self, batch_size, device):
-        """This method set the first tokens to be decoder_input_tokens during search."""
-        return torch.tensor([self.decoder_input_tokens] * batch_size).to(device)
-
-    def permute_mem(self, memory, index):
-        """Memory permutation during beamsearch."""
-        memory = torch.index_select(memory, dim=0, index=index)
-        return memory
-
-    def forward_step(self, inp_tokens, memory, enc_states, enc_lens):
-        """Performs a step in the implemented beamsearcher."""
-        memory = _update_mem(inp_tokens, memory)
-
-        # WARNING: the max_decode_ratio need to be under 448 because
-        #  of positional encoding
-        dec_out, attn = self.model.forward_decoder(enc_states, memory)
-        log_probs = self.softmax(dec_out[:, -1])
-
-        return log_probs, memory, attn
-
-    def change_max_decoding_length(self, min_decode_steps, max_decode_steps):
-        """set the minimum/maximum length the decoder can take."""
-        return (
-            int(self.min_decode_ratio * self.max_length),
-            int(self.max_decode_ratio * self.max_length),
-        )
-
-
-class S2STransformerGreedySearch(S2SGreedySearcher):
-    """This class implements the greedy decoding
-    for Transformer.
-
-    Arguments
-    ---------
-    modules : list with the following one:
-        model : torch.nn.Module
-            A TransformerASR model.
-        seq_lin : torch.nn.Module
-            A linear output layer for the seq2seq model.
-    temperature : float
-        Temperature factor applied to softmax. It changes the probability
-        distribution, being softer when T>1 and sharper with T<1.
-    **kwargs
-        Arguments to pass to S2SGreedySearcher
-    """
-
-    def __init__(self, modules, temperature=1.0, **kwargs):
-        super(S2SGreedySearcher, self).__init__(**kwargs)
-
-        self.model = modules[0]
-        self.fc = modules[1]
-        self.softmax = torch.nn.LogSoftmax(dim=-1)
-
-        self.temperature = temperature
-
-    def reset_mem(self, batch_size, device):
-        """Needed to reset the memory during greedy search."""
-        return None
-
-    def forward_step(self, inp_tokens, memory, enc_states, enc_lens):
-        """Performs a step in the implemented greedy searcher."""
-        memory = _update_mem(inp_tokens, memory)
-        pred, attn = self.model.decode(memory, enc_states, enc_lens)
-        prob_dist = self.softmax(self.fc(pred) / self.temperature)
-        return prob_dist[:, -1, :], memory, attn
-
-
-class S2SWhisperBeamSearch(S2SBeamSearcher):
+class S2SWhisperBeamSearcher(S2SBeamSearcher):
```

Comment:
I think in this context this is fine to have duplicated methods. I would prefer this since most people will copy paste and do their own sauce with the searcher in case they want to modify it. It improves the readability imho.